definitions/npm/react-router-dom_v6.x.x/flow_v0.104.x-/test_react-router-dom.js

```diff
@@ -484,5 +485,54 @@ describe("react-router-dom", () => {
         sensitive: 'foo',
       });
     });
+
+    it('useMatches', () => {
+      type Params<Key: string> = {
+        +[key: Key]: string | void,
+      };
+
+      type Matches = Array<{|
```

Comment:
Probably interesting to have the type exposed from the lib as well?

Reply:
That was my first though but it could mislead with other types and it's neither exported for typescript since the returned value is computed in `useMatch`: https://github.com/remix-run/react-router/blob/main/packages/react-router/lib/hooks.tsx#L739-L760

Reply:
should I create this new type so?

Reply:
Yeah, maybe not. 

The TS decalration file is as: 
```
export declare function useMatches(): {
    id: string;
    pathname: string;
    params: Params<string>;
    data: unknown;
    handle: unknown;
}[];
```